Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -239,45 +202,117 @@ void CGPathAddLineToPoint(CGMutablePathRef path, const CGAffineTransform* m, flo
 
     if (pathObj->_count + 1 >= pathObj->_max) {
         pathObj->_max += 32;
-        pathObj->_components = (pathComponent*)IwRealloc(pathObj->_components, pathObj->_max * sizeof(pathComponent));
+        pathObj->_components = (CGPathElement*)IwRealloc(pathObj->_components, pathObj->_max * sizeof(CGPathElement));
     }
 
-    pathObj->_components[pathObj->_count].type = pathComponentLineTo;
-    pathObj->_components[pathObj->_count].point.x = x;
-    pathObj->_components[pathObj->_count].point.y = y;
+    pathObj->_components[pathObj->_count].type = kCGPathElementAddLineToPoint;
+
+    pathObj->_components[pathObj->_count].points = (CGPoint*)IwCalloc(1, sizeof(CGPoint));
+    pathObj->_components[pathObj->_count].points[0] = { x, y };
 
     pathObj->_count++;
 }
 
+CGFloat _CGPathControlPointOffsetMultiplier(CGFloat angle) {
+    return (4.0f / 3.0f) * tan(angle / 4.0f);
+}
+
 /**
  @Status Caveat
  @Notes transform property not supported
 */
 void CGPathAddArcToPoint(CGMutablePathRef path, const CGAffineTransform* m, float x1, float y1, float x2, float y2, float radius) {
-    if (m) {
-        assert(0);
+    bool isEmpty = CGPathIsEmpty(path);
+
+    if (isEmpty) {
+        return;
     }
 
-    CGPathRef pathObj = path;
+    CGPoint curPathPosition = CGPathGetCurrentPoint(path);
 
-    if (pathObj->_count + 1 >= pathObj->_max) {
-        pathObj->_max += 32;
-        pathObj->_components = (pathComponent*)IwRealloc(pathObj->_components, pathObj->_max * sizeof(pathComponent));
+    double x0, y0;
+    double dx0, dy0, dx2, dy2, xl0, xl2;
+    double san, n0x, n0y, n2x, n2y, t;
+
+    x0 = curPathPosition.x;
+    y0 = curPathPosition.y;
+
+    dx0 = x0 - x1;
+    dy0 = y0 - y1;
+    xl0 = sqrt(dx0 * dx0 + dy0 * dy0);
```

Comment:
Actually would be nice to make dx0 deltaX10. I was confused why there was no dx1 / dy1 but that is because a single point was chosen for the deltas. 

---
In reply to: [68683240](https://github.com/Microsoft/WinObjC/pull/559#discussion_r68683240) [](ancestors = 68683240)